### PR TITLE
fix: [NPM] add check for valid IPV4 addresses in TranslatePolicy

### PR DESCRIPTION
**Pull request by ck319**
**Reason for Change**:

This fixes an issue where unsupported IPV6 addresses were not being validated at the translate step, causing an error when attempting to add them to the dataplane.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests

**Notes**:
Add check for valid IPV4 addresses in the translate policy step

### CODE REVIEW

File: npm/pkg/controlplane/translation/translatePolicy.go [hunk 1]
```diff
@@ -27,7 +27,8 @@ var (
 	// ErrUnsupportedExceptCIDR is returned when Except CIDR block translation feature is used in windows.
 	ErrUnsupportedExceptCIDR = errors.New("unsupported Except CIDR block translation features used on windows")
 	// ErrUnsupportedSCTP is returned when SCTP protocol is used in windows.
-	ErrUnsupportedSCTP = errors.New("unsupported SCTP protocol used on windows")
+	ErrUnsupportedSCTP    = errors.New("unsupported SCTP protocol used on windows")
+	ErrInvalidIPV6Address = errors.New("invalid IPV6 address")
```
Comment (huntergregory) [inline npm/pkg/controlplane/translation/translatePolicy.go:31]: nitpick: seems this error will be thrown for any IPv6 or for an invalid IPv4. Maybe "unsupported" is a better word for IPv6?

**Review by huntergregory — CHANGES_REQUESTED** [review]
left some comments. Could you also add a short description of the current issue and how your PR fixes it?

File: npm/pkg/controlplane/controllers/v2/networkPolicyController.go [hunk 2]
```diff
@@ -296,7 +296,7 @@ func (c *NetworkPolicyController) syncAddAndUpdateNetPol(netPolObj *networkingv1
 
 		klog.Errorf("Failed to translate podSelector in NetworkPolicy %s in namespace %s: %s", netPolObj.ObjectMeta.Name, netPolObj.ObjectMeta.Namespace, err.Error())
 		// The exec time isn't relevant here, so consider a no-op.
-		return metrics.NoOp, errNetPolTranslationFailure
+		return metrics.NoOp, nil
```
[the remark this answers is not in the document]
Comment (ck319) [reply npm/pkg/controlplane/controllers/v2/networkPolicyController.go:299]: updated

File: npm/pkg/controlplane/translation/translatePolicy.go (hunk 1 above)
Comment (ck319) [reply npm/pkg/controlplane/translation/translatePolicy.go]: updated

File: npm/pkg/controlplane/translation/translatePolicy.go [hunk 3]
```diff
@@ -551,12 +552,31 @@ func egressPolicy(npmNetPol *policies.NPMNetworkPolicy, netPolName string, egres
...
-// TranslatePolicy traslates networkpolicy object to NPMNetworkPolicy object
-// and return the NPMNetworkPolicy object.
+// TranslatePolicy translates networkpolicy object to NPMNetworkPolicy object
+// and returns the NPMNetworkPolicy object.
 func TranslatePolicy(npObj *networkingv1.NetworkPolicy) (*policies.NPMNetworkPolicy, error) {
 	netPolName := npObj.Name
 	npmNetPol := policies.NewNPMNetworkPolicy(netPolName, npObj.Namespace)
 
+	// check if IPs are valid IPV4 addresses
+	for _, egress := range npObj.Spec.Egress {
```
[the remark this answers is not in the document]
Comment (ck319) [reply npm/pkg/controlplane/translation/translatePolicy.go]: updated

**Review by huntergregory — APPROVED** [review]

**Review by huntergregory — APPROVED** [review]

**Pull request merged**